test/jdk/java/awt/PopupMenu/PopupMenuLocation.java

```diff
@@ -106,22 +123,36 @@ private void show(MouseEvent e) {
     }
 
     private static void openPopup(final Frame frame) throws Exception {
-        Robot robot = new Robot();
-        robot.setAutoDelay(200);
         robot.waitForIdle();
         Point pt = frame.getLocationOnScreen();
-        robot.mouseMove(pt.x + frame.getWidth() / 2, pt.y + 50);
+        int x = pt.x + frame.getWidth() / 2;
+        int y = pt.y + 50;
+        robot.mouseMove(x, y);
         robot.mousePress(InputEvent.BUTTON3_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON3_DOWN_MASK);
-        int x = pt.x + frame.getWidth() / 2;
-        int y = pt.y + 130;
+        robot.waitForIdle();
+         y = y+50;
         robot.mouseMove(x, y);
         robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
         robot.waitForIdle();
         if (!action) {
-            throw new RuntimeException();
+            captureScreen();
+            throw new RuntimeException(
+                    "Failed, Not received the PopupMenu ActionEvent yet on " +
+                    "frame= "+frame+", isFocused = "+frame.isFocused());
         }
         action = false;
     }
+
+    private static void captureScreen() {
+        try {
+            ImageIO.write(robot.createScreenCapture(currentScreenBounds), "png",
+                          new File("screen1.png"));
```

Comment:
Is screen1.png the best name for the image? May be misleading if there are multiple screens, especially since this test iterates multiple locations.

Reply:
It is definitely not. Either keep track of the current screen or use a generic `screen.png` file name.

I would also suggest, wrapping all the parameters to `write`:
```java
            ImageIO.write(robot.createScreenCapture(screenBounds),
                          "png",
                          new File("screen1.png"));
```
This way the `"png"` doesn't get “lost”.